src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -50,19 +53,39 @@ class SDCPUThread
     class SDCPUExecContext : public ExecContext
     {
       protected:
-        SDCPUThread& backingState;
+        InflightInst& backingState;
         
       public:
         SDCPUExecContext(SDCPUThread& backing_state):
             backingState(backing_state)
         { }
     };
 
-    class DynamicInst
+    class InflightInst
     {
-        StaticInst& instructionRef;
-        // TODO instruction tag
-        // TODO instruction results
+      public:
+        enum Status
+        {
+            Invalid = 0,
+            Waiting, // Can't be executed because dependencies remain
+            Executing, // Requirements met, but waiting for results
+            Completed // Results have been received, but not yet committed.
+        };
+
+        // Where am I in completing this instruction?
+        Status status;
+
+        // What is this instruction I'm executing?
+        InstSeqNum tag;
+        StaticInstPtr instRef;
+
+        // How does this instruction influence my system?
+        vector<InstResult> results;
+        vector<TheISA::CCReg> ccResults;
+        Fault fault;
+
+        // TODO deal with macroops?
+        // TODO deal with memory operations, a la LSQ?
```

Comment:
You'll likely need an effective address and a `MemoryRequest`-like reference.

Though, maybe some of this is more appropriate in the ExecutionContext above. I'm not sure what the difference between that class and this class is.

Either in this class or `ExecContext` there are a few more things you're going to need.

1. Place to hold all of the source data. When this instruction is decoded you'll need to read the architectural registers that aren't busy *at that time*. Similarly, if the architectural register is busy, then you'll need a place for the preceding instruction to write the data when it's done.
2. A place to hold the data you're going to store.
3. A place to hole information for the memory request (like mentioned above).